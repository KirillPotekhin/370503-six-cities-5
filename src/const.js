export const SortingOption = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const SizePreviewImage = {
  WIDTH_PREVIEW_DEFAULT: `260`,
  HEIGHT_PREVIEW_DEFAULT: `220`,
  WIDTH_PREVIEW_FOR_FAVORITES: `150`,
  HEIGHT_PREVIEW_FOR_FAVORITES: `110`,
};

export const ReviewRatingTitle = {
  PERFECT: `perfect`,
  GOOD: `good`,
  NOT_BAD: `not bad`,
  BEADLY: `beadly`,
  TERRIBLY: `terribly`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
