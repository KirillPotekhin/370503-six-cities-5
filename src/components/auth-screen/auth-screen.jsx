import React from "react";
import {Link} from "react-router-dom";
import applicationPropTypes from "../../application-prop-types";
import {AppRoute} from "../../const";
import withAuthData from "../../hocs/with-auth-data/with-auth-data";

const AuthScreen = (props) => {
  const {city} = props;
  return (
    <div className="page page--gray page--login">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <Link to={AppRoute.ROOT} className="header__logo-link">
                <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41" />
              </Link>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <a className="header__nav-link header__nav-link--profile" href="#">
                    <div className="header__avatar-wrapper user__avatar-wrapper">
                    </div>
                    <span className="header__login">Sign in</span>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>

      <main className="page__main page__main--login">
        <div className="page__login-container container">
          <section className="login">
            <h1 className="login__title">Sign in</h1>
            <form className="login__form form" action="#" method="post" onSubmit={props.onSubmitForm}>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">E-mail</label>
                <input
                  className="login__input form__input"
                  type="email" name="email"
                  placeholder="Email"
                  required
                  value={props.login}
                  onChange={props.onChangeField(`login`)}
                />
              </div>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">Password</label>
                <input
                  className="login__input form__input"
                  type="password"
                  name="password"
                  placeholder="Password"
                  required
                  value={props.password}
                  onChange={props.onChangeField(`password`)}
                />
              </div>
              <button className="login__submit form__submit button" type="submit">Sign in</button>
            </form>
          </section>
          <section className="locations locations--login locations--current">
            <div className="locations__item">
              <a className="locations__item-link" href="#">
                <span>{city.name}</span>
              </a>
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

AuthScreen.propTypes = {
  city: applicationPropTypes.city,
  login: applicationPropTypes.login,
  password: applicationPropTypes.password,
  onSubmitForm: applicationPropTypes.onSubmitForm,
  onChangeField: applicationPropTypes.onChangeField,

};

export {AuthScreen};
export default withAuthData(AuthScreen);
