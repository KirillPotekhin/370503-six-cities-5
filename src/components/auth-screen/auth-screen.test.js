import React from "react";
import renderer from "react-test-renderer";
import {AuthScreen} from "./auth-screen";
import {Route, Switch, Router as BrowserRouter} from "react-router-dom";
import {AppRoute} from "../../const";
import browserHistory from "../../browser-history";

const noop = () => () => {};

const city = {
  location: {
    latitude: 52.3909553943508,
    longitude: 4.85309666406198,
    zoom: 12,
  },
  name: `Amsterdam`,
};

it(`Should AuthScreen render correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter history={browserHistory}>
          <Switch>
            <Route exact path={AppRoute.LOGIN}>
              <AuthScreen
                city={city}
                login={`hi@MediaList.ru`}
                password={`sdfsdfsf`}
                onSubmitForm={noop}
                onChangeField={noop}
              />
            </Route>
          </Switch>
        </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
