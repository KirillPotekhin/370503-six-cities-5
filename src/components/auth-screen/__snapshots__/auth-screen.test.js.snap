// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AuthScreen render correctly 1`] = `null`;
